compiler/ccgexprs.nim

```diff
@@ -869,16 +871,31 @@ proc genFieldCheck(p: BProc, e: PNode, obj: Rope, field: PSym) =
     v.r.add(".")
     v.r.add(disc.sym.loc.r)
     genInExprAux(p, it, u, v, test)
-    let msg = genFieldDefect(field, disc.sym)
-    let strLit = genStringLiteral(p.module, newStrNode(nkStrLit, msg))
-    if op.magic == mNot:
-      linefmt(p, cpsStmts,
-              "if ($1){ #raiseFieldError($2); $3}$n",
-              [rdLoc(test), strLit, raiseInstr(p)])
-    else:
-      linefmt(p, cpsStmts,
-              "if (!($1)){ #raiseFieldError($2); $3}$n",
-              [rdLoc(test), strLit, raiseInstr(p)])
+    var msg = toFileLineCol(p.config, e.info)
+      # shows the most relevant lineinfo even with -d:release;
+      # should have no runtime cost (unlike -d:stacktrace)
+      # this could be added to all/most chcks.nim errors, and perhaps
+      # can be disabled with a compile flag.
+    msg.add " " & genFieldDefect(p.config, field, disc.sym)
+
+    template newLitRope(s: string): untyped = genStringLiteral(p.module, newStrNode(nkStrLit, s))
+    let strLit = newLitRope(msg)
+    ## discriminant check
+    template fun(code) = linefmt(p, cpsStmts, code, [rdLoc(test)])
+    if op.magic == mNot: fun("if ($1) ") else: fun("if (!($1)) ")
+
+    ## raiseFieldError2 on failure
+    if optTinyRtti in p.config.globalOptions: # optNimV2 disappeared in 61ea85687c2950bb40c23a1a7cd2c13473bd9662
+      const code = "{ #raiseFieldError2($1, $2); $3} $n"
+      linefmt(p, cpsStmts, code, [strLit, newLitRope("(repr unavailable in newruntime)"), raiseInstr(p)])
```

Comment:
repr is now available in --newruntime/arc/orc so it should be used.

Reply:
that doesn't help much for RTTI though, all i can get is the `ord`, not the corresponding symbol string; the latest commits now show at least the ord instead of `repr unavailable in newruntime` for newruntime (eg gc:arc)